Interface.C

```diff
@@ -275,6 +280,75 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
             }
         }
     }
+
+    else if (locationType_ == LocationType::volumeCenters)
+    {
+        // The volume coupling implementation is based on the coupling with faceCenters
+        // as the volume coupling was tested and developed on a previous development where faceNodes
+        // did not work properly
+        //
+        // The volume coupling implementation considers the mesh points in the volume and
+        // on the boundary patches in order to take the boundary conditions into account
+
+        // Get the number of (volume centered) mesh points in the volume
+        numDataLocations_ = mesh.C().size();
+
+        // Count the data locations for all the patches
+        // and add those to the previously determined number of mesh points in the volume
+        for (uint j = 0; j < patchIDs_.size(); j++)
+        {
+            numDataLocations_ +=
+                mesh.boundaryMesh()[patchIDs_.at(j)].faceCentres().size();
+        }
+        DEBUG(adapterInfo("Number of coupling volumes: " + std::to_string(numDataLocations_)));
+
+        // Array of the mesh vertices.
+        // One mesh is used for all the patches and each vertex has 3D coordinates.
+        double vertices[dim_ * numDataLocations_];
+
+        // Array of the indices of the mesh vertices.
+        // Each vertex has one index, but three coordinates.
+        vertexIDs_ = new int[numDataLocations_];
+
+        // Initialize the index of the vertices array
+        int verticesIndex = 0;
+
+        // Get the locations of the volume centered mesh vertices
+        const vectorField& CellCenters = mesh.C();
+
+        for (uint i = 0; i < CellCenters.size(); i++)
```

Comment:
I am getting a compiler warning with OpenFOAM v2212 and GCC 11.3:

```
g++ -std=c++14 -m64 -pthread -DOPENFOAM=2212 -DWM_DP -DWM_LABEL_SIZE=32 -Wall -Wextra -Wold-style-cast -Wnon-virtual-dtor -Wno-unused-parameter -Wno-invalid-offsetof -Wno-attributes -Wno-unknown-pragmas -O3  -DNoRepository -ftemplate-depth-100 -fdiagnostics-color=always -I/usr/lib/openfoam/openfoam2212/src/finiteVolume/lnInclude -I/usr/lib/openfoam/openfoam2212/src/meshTools/lnInclude -I/usr/lib/openfoam/openfoam2212/src/transportModels/ -I/usr/lib/openfoam/openfoam2212/src/transportModels/incompressible/lnInclude -I/usr/lib/openfoam/openfoam2212/src/transportModels/compressible/lnInclude -I/usr/lib/openfoam/openfoam2212/src/transportModels/twoPhaseMixture/lnInclude -I/usr/lib/openfoam/openfoam2212/src/transportModels/interfaceProperties/lnInclude -I/usr/lib/openfoam/openfoam2212/src/transportModels/immiscibleIncompressibleTwoPhaseMixture/lnInclude -I/usr/lib/openfoam/openfoam2212/src/thermophysicalModels/basic/lnInclude -I/usr/lib/openfoam/openfoam2212/src/TurbulenceModels/turbulenceModels/lnInclude -I/usr/lib/openfoam/openfoam2212/src/TurbulenceModels/compressible/lnInclude -I/usr/lib/openfoam/openfoam2212/src/TurbulenceModels/incompressible/lnInclude -I/usr/lib/openfoam/openfoam2212/src/triSurface/lnInclude -I/home/gc/repos/precice/precice/src -I../  -iquote. -IlnInclude -I/usr/lib/openfoam/openfoam2212/src/OpenFOAM/lnInclude -I/usr/lib/openfoam/openfoam2212/src/OSspecific/POSIX/lnInclude   -fPIC -c FSI/FSI.C -o Make/linux64GccDPInt32Opt/FSI/FSI.o
Interface.C: In member function ‘void preciceAdapter::Interface::configureMesh(const Foam::fvMesh&, const string&, const string&)’:
Interface.C:362:28: warning: comparison of integer expressions of different signedness: ‘uint’ {aka ‘unsigned int’} and ‘Foam::label’ {aka ‘int’} [-Wsign-compare]
  362 |         for (uint i = 0; i < CellCenters.size(); i++)
      |            
```

(not sure why I get different line numbers)

Reply:
It seems that `CellCenters.size()` is of type `Foam::label` which is an `int64_t`/`int32_t` (signed) under the hood. Solved the problem by casting for now...

Reply:
Why not just `auto`? We can use C++11 features.

Reply:
Thanks, that worked!